### PR TITLE
[MPS] Fix roi_align backward gradient over-accumulation

### PR DESCRIPTION
**Pull request by malfet**
The backward kernel ran a grid-stride loop dispatched over multiple threadgroups but strided by only one threadgroup's width, so every threadgroup re-processed overlapping output elements and atomic-added each gradient repeatedly.
This is harmless for the forward (idempotent writes, fixed for perf in #9100) but the backward over-accumulated input gradients by a factor that grows with the RoI count (~100x at 512 RoIs), making detection training diverge to NaN on MPS.
Fix mirrors #9100: one thread per pooled-output element via dispatchThreads.

Test Plan:
  python -m pytest test/test_ops.py -k "test_backward_mps_consistency"

Authored with assistance from Claude.

### CODE REVIEW

**Review by NicolasHug — APPROVED** [review]
Thanks @malfet , this LGTM, I'll merge once the CI is kind enough to be green (macos jobs are failing on a new conda issue right now, I'm looking into it)

**Comment by NicolasHug** [comment]
I was able to get the CI back to green, I'll be able to merge this once you sign the CLA @malfet  (🥲!)

**Comment by jveitchmichaelis** [comment]
I believe this fix should also be applied to `roi_pool`, `ps_roi_align` and `ps_roi_pool` which have the same issue in forward + backward passes.

e.g.

``` python
import torch
from torchvision.ops import roi_pool

def grad_sum(dev):
    x = torch.ones(1, 64, 8, 8, device=dev, requires_grad=True)
    rois = torch.tensor([[0.0, 0, 0, 7, 7]], device=dev) # 1 box 
    roi_pool(x, rois, [4, 4]).sum().backward()
    return x.grad.sum()

assert torch.isclose(grad_sum("cpu"), grad_sum("mps"))   # Fails, 1024.0  != 1536.0
```
